src/mbgl/style/style_parser.cpp

```diff
@@ -242,16 +241,21 @@ void StyleParser::parseSources(const JSValue& value) {
     }
 }
 
-std::unique_ptr<mapbox::geojsonvt::GeoJSONVT> StyleParser::parseGeoJSON(const JSValue& value) {
+std::unique_ptr<mapbox::geojsonvt::GeoJSONVT> StyleParser::parseGeoJSON(const JSValue& value, std::unique_ptr<SourceInfo>& info) {
     using namespace mapbox::geojsonvt;
 
+    Options options;
+    options.buffer = util::EXTENT / util::tileSize * 128;
+    options.extent = util::EXTENT;
+    info->maxZoom = options.maxZoom;
+
     try {
-        return std::make_unique<GeoJSONVT>(Convert::convert(value, 0));
+        return std::make_unique<GeoJSONVT>(Convert::convert(value, 0), options);
```

Comment:
Does `GeoJSONVT` provide access to the `options` passed to the constructor? If so, you could avoid passing `SourceInfo` here and instead have the caller set `maxZoom` based on the options from the `GeoJSONVT` return value.